services/motion/explore/explore.go

```diff
@@ -0,0 +1,571 @@
+
+	// obstacleDetectors
+	obstacleDetectors, err := ms.createObstacleDetectors(motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create kinematic base
+	kb, err := ms.createKinematicBase(ctx, componentName, motionCfg)
+	if err != nil {
+		return false, err
+	}
+
+	// Create motionplan plan
+	plan, err := ms.createMotionPlan(ctx, kb, destination, worldState, extra)
+	if err != nil {
+		return false, err
+	}
+
+	// Start background processes
+	cancelCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Start polling for obstacles
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+		ms.checkForObstacles(cancelCtx, obstacleDetectors, kb, plan, motionCfg.ObstaclePollingFreqHz)
+	}, ms.backgroundWorkers.Done)
+
+	// Start executing plan
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+		ms.executePlan(cancelCtx, kb, plan)
+	}, ms.backgroundWorkers.Done)
+
+	for {
+		// this ensures that if the context is cancelled we always return early at the top of the loop
+		if err := ctx.Err(); err != nil {
+			return false, err
+		}
+
+		select {
+		// if context was cancelled by the calling function, error out
+		case <-ctx.Done():
+			return false, ctx.Err()
+
+		// once execution responds: return the result to the caller
+		case resp := <-ms.executionResponseChan:
+			ms.logger.Debugf("execution completed: %s", resp)
+			return resp.success, resp.err
+
+		// if the checkPartialPlan process hit an error return it, otherwise exit
+		case resp := <-ms.obstacleResponseChan:
+			ms.logger.Debugf("obstacle response: %s", resp)
+			if resp.err != nil {
+				return resp.success, resp.err
+			}
+			if resp.success {
+				return resp.success, nil
+			}
+		}
+	}
+}
+
+// checkForObstacles will continuously monitor the generated transient worldState for obstacles in the given
+// motionplan plan. A response will be sent through the channel if an error occurs, the motionplan plan
+// completes or an obstacle is detected in the given range.
+func (ms *explore) checkForObstacles(
+	ctx context.Context,
+	obstacleDetectors []obstacleDetectorObject,
+	kb kinematicbase.KinematicBase,
+	plan motionplan.Plan,
+	obstaclePollingFrequencyHz float64,
+) {
+	// Constantly check for obstacles in path at desired obstacle polling frequency
+	ticker := time.NewTicker(time.Duration(int(1000/obstaclePollingFrequencyHz)) * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			currentPose := spatialmath.NewZeroPose()
+
+			// Look for new transient obstacles and add to worldState
+			worldState, err := ms.generateTransientWorldState(ctx, obstacleDetectors)
+			if err != nil {
+				ms.obstacleResponseChan <- moveResponse{err: err}
+				return
+			}
+
+			// Check motionplan plan for transient obstacles
+			collisionPose, err := motionplan.CheckPlan(
+				kb.Kinematics(),
+				plan,
+				worldState,
+				ms.frameSystem,
+				currentPose,
+				[]referenceframe.Input{{Value: 0}, {Value: 0}},
```

Comment:
This should not be hardcoded

Reply:
Move to default at top of file.

Reply:
Since we are using a nil localizer, we can't get the `Inputs` from the kinematic base

Reply:
Implemented non-hardcoded version based on comment below